cpp/src/nodes/each.cpp

```diff
@@ -0,0 +1,37 @@
+#include "each.hpp"
+#include <vector>
+
+Each::Each(std::string collection_name, std::string val_name, std::string key_name):
+  collection_name(collection_name),
+  val_name(val_name),
+  key_name(key_name)
+{}
+
+std::string Each::to_html(NodePtr body, Json &context) const {
```

Comment:
This method is kinda big, and hard to read - do you think we could break this up into a few smaller private methods belonging to the `Each` class?